include/extractor/scripting_environment.hpp

```diff
@@ -36,6 +41,13 @@ class RestrictionParser;
 struct ExtractionNode;
 struct ExtractionWay;
 
+using index_sparse_type = osmium::index::map::SparseFileArray<osmium::unsigned_object_id_type, osmium::Location>;
+using index_dense_type = osmium::index::map::DenseFileArray<osmium::unsigned_object_id_type, osmium::Location>;
```

Comment:
You let the underlying system take care of swapping (which heavily degrades performance).